assets/javascripts/running.js

```diff
@@ -268,4 +288,435 @@ function pauseLiveView() {
+    });
+
+    // register handlers for controls
+    $('#developer-pause-at-module').on('change', handleModuleToPauseAtSelected);
+
+    // find URLs for web socket connections
+    developerMode.develWsUrl = panel.data('developer-url');
+    developerMode.statusOnlyWsUrl = panel.data('status-only-url');
+
+    updateDeveloperPanel();
+    setupWebsocketConnection();
+}
+
+// updates the developer panel, must be called after modifying developerMode
+function updateDeveloperPanel() {
+    // hide/show elements according to data-hidden and data-visible attributes
+    var developerModeElements = $('.developer-mode-element');
+    developerModeElements.each(function(index) {
+        var element = $(this);
+        var visibleOn = element.data('visible-on');
+        var hiddenOn = element.data('hidden-on');
+        var hide = ((hiddenOn && developerMode.prop(hiddenOn))
+                 || (visibleOn && !developerMode.prop(visibleOn)));
+        if (hide) {
+            element.hide();
+        } else if (element.hasClass('btn')) {
+            element.css('display', 'inline-block');
+        } else {
+            element.show();
+        }
+    });
+
+    // set panel visibility
+    var panel = $('#developer-panel');
+    if (!testStatus.running) {
+        // hide entire panel if test is not running anymore
+        panel.hide();
+        return;
+    }
+    panel.show();
+
+    // toggle panel body if its current state doesn't match developerMode.panelExpanded
+    var panelBody = panel.find('.card-body');
+    if (developerMode.panelExpanded !== developerMode.panelActuallyExpanded) {
+        developerMode.panelActuallyExpanded = developerMode.panelExpanded;
+        panelBody.toggle(200);
+    }
+
+    // find modules
+    var moduleToPauseAtSelect = $('#developer-pause-at-module');
+    var moduleToPauseAtOptions = moduleToPauseAtSelect.find('option');
+    var modules = moduleToPauseAtOptions.map(function() {
+        var option = $(this);
+        var category = option.parent('optgroup').attr('label');
+        return category ? (category + '-' + option.val()) : option.val();
+    }).get();
+    var currentModuleIndex = modules.indexOf(developerMode.currentModule);
+    var toPauseAtIndex = modules.indexOf(developerMode.moduleToPauseAt);
+    if (toPauseAtIndex < 0) {
+        toPauseAtIndex = 0;
+    }
+    var moduleToPauseAtStillAhead = (developerMode.moduleToPauseAt
+        && toPauseAtIndex > currentModuleIndex);
+
+    // update status info
+    var statusInfo = 'unknown';
+    if (developerMode.isPaused) {
+        statusInfo = 'paused';
+    } else if (moduleToPauseAtStillAhead) {
+        statusInfo = 'will pause at module ' + developerMode.moduleToPauseAt;
+    } else {
+        statusInfo = 'usual test execution';
+    }
+    if (developerMode.currentModule && !moduleToPauseAtStillAhead) {
+         statusInfo += ', at ' + developerMode.currentModule;
+    }
+    $('#developer-status-info').text(statusInfo);
```

Comment:
I would actually put the name of the module in bold when the test execution is set to be paused.